frontend/packages/console-app/src/components/user-preferences/theme/usePreferredTheme.ts

```diff
@@ -0,0 +1,11 @@
+import { Dispatch, SetStateAction } from 'react';
+import { useUserSettings } from '@console/shared';
+
+export const usePreferredTheme = (): [string, Dispatch<SetStateAction<string>>, boolean] => {
+  const [preferredTheme, setPreferredTheme, preferredThemeLoaded] = useUserSettings<string>(
```

Comment:
We should cache the last known value in localStorage to avoid a flicker when loading console (and then update with the user preferences value when it loads).

Reply:
Hey, I tested this with "fast 3g" network throttling and you can easily see how the UI switches from black to white for some seconds, back to black. I would suggest adding this to index.html so that it's set as soon as the HTML is fetched. Wdyt Sam? Our JS load would take to long to "fix flickering"

Reply:
@jerolimov Thanks for testing this. I agree we should add the default class to index.html to avoid the flicker. It should only be a couple lines of JS.